src/Agent.cs

```diff
@@ -26,6 +26,14 @@ public class Agent
         private readonly Collector _collector;
         private static readonly ILog _log = LogManager.GetCurrentClassLogger();
 
+        public int MessageCount
+        {
+            get
+            {
+                return _collector.MessageCount;
```

Comment:
`Flush` and tests use this function to determine when all messages have been sent but it can be off by one due to _currentCommand not being flushed. How about something like this:

```sharp
_collector.MessageCount + (_currentCommand ? 1 : 0);
```